### ISSUE TITLE
Discussion on real-ranged attacks preshow

### ISSUE THREAD

**Issue by hladkyjakub** [wesnoth/wesnoth#9265]
Hi.
I'm working on a real-ranged attacks menu.
When you want to move&close-attack a unit that is further than on an adjacent hex, while you have a real-ranged attack, you don't get an option to attack from close range (you have to move the unit manually). Which in tests proved as really tedious.
So I want to make the attack menu, in a way, it shows all unit attacks (from different locations).
I made the thing in Lua and got good results (except for the pathfinding, which will be finished later)
The question is: How to mark all possible attack hexes in the attack preshow (when attacker is selected and the target is being hovered over)?
Showing two unit images over two hexes seems weird

*note: this won't affect anything mainline.
*note2: the current system isn't ready to show more paths at once... if somebody could give me some advice on how to show more paths and/or attack source hexes at once, it would be very appreciated.

**Comment by hladkyjakub** [comment]
Got it working for one tile, viz screenshots.
![Screenshot_20240827_190618](https://github.com/user-attachments/assets/1ed83cac-2a5e-47c1-bc96-ecb399607578)
![Screenshot_20240827_190726](https://github.com/user-attachments/assets/95f4c3c3-1979-41d0-8d95-a0bd79ca7b4e)
Is the graphic ok?
(When hovered over)

**Comment by knyghtmare** [comment]
Got any video links of showing the real-ranged feature?

**Comment by hladkyjakub** [comment]
> Got any video links of showing the real-ranged feature?

I'm unable to provide it right now, I've uploaded the Real Ranged Attacks add-on on 1.18 and 1.19 servers.
You can see the functionality there, tho it misses propper graphic and has bad auto-path find.

**Comment by CelticMinstrel** [comment]
Hmm. I think we should add a component to extent the arrow, sorta like this mock-up.

![image](https://github.com/user-attachments/assets/af096e30-3316-4a9a-9577-4033231dd178)

**Comment by hladkyjakub** [comment]
> Hmm. I think we should add a component to extent the arrow, sorta like this mock-up.
> 
> ![image](https://github.com/user-attachments/assets/af096e30-3316-4a9a-9577-4033231dd178)

I'm not against it, but how to do this once multiple attack hexes will be highlighted?
(Do to unit having more weapons with different max/min ranges)

**Comment by CelticMinstrel** [comment]
Do you mean if there are multiple ways they could attack the target hex? For example, they could move to a space 3 hexes away and use an attack with a max range of 3, or move one hex closer and use an attack with a max range of 2? In that case I'd say it should choose the attack with the highest max_range. If they specifically want to use a different attack, they can move and attack as separate actions.

**Comment by hladkyjakub** [comment]
You're right, but I've found extremelly tidious for me to move and then attack with units like spearman from close.
So united menu for all ranges is better to be there.

**Comment by CelticMinstrel** [comment]
Even if the menu _is_ going to show all ranges (which seems excessive to me), I think it's reasonable to just choose the max range for the on-map visualization.

**Comment by hladkyjakub** [comment]
> Even if the menu _is_ going to show all ranges (which seems excessive to me), I think it's reasonable to just choose the max range for the on-map visualization.

I agree with you, in cases one attack can be used.

But what to do in this example:
Spearman, with javelin with max/min_range =3-2
You can't tell which attack is going to be used, how to highlight now?

**Comment by CelticMinstrel** [comment]
I don't understand your example, so I can't come up with an answer to the question.

**Comment by hladkyjakub** [comment]
> I don't understand your example, so I can't come up with an answer to the question.

Most of the time you  want to attaxk melee with him.

**Comment by CelticMinstrel** [comment]
That doesn't really make sense to me. If you can attack without retaliation, isn't that nearly always better?

**Comment by hladkyjakub** [comment]
> That doesn't really make sense to me. If you can attack without retaliation, isn't that nearly always better?

Sorry, for being unclear, I mean agains enemy archer, whichs bow has same max/min range as javelin.
